src/mono/mono/metadata/object.c

```diff
@@ -6744,16 +6744,36 @@ mono_string_is_interned_lookup (MonoStringHandle str, gboolean insert, MonoError
 	res = (MonoString *)mono_g_hash_table_lookup (ldstr_table, MONO_HANDLE_RAW (str));
 	if (res)
 		MONO_HANDLE_ASSIGN_RAW (s, res);
-	else
+	else {
 		mono_g_hash_table_insert_internal (ldstr_table, MONO_HANDLE_RAW (s), MONO_HANDLE_RAW (s));
+
+#ifdef HOST_WASM
+		mono_set_string_interned_internal ((MonoObject *)MONO_HANDLE_RAW (s));
+#endif
+	}
 	ldstr_unlock ();
 	return s;
 }
 
+#ifdef HOST_WASM
+/**
+ * mono_string_instance_is_interned:
+ * Searches the interned string table for the provided string instance.
+ * \param str String to probe
+ * \returns TRUE if the string is interned, FALSE otherwise.
+ */
+int
+mono_string_instance_is_interned (MonoString *str)
```

Comment:
I think this still needs a declaration in `object-internals.h` (or some place) to avoid the "no previous declaration" warning from the C compiler.

Reply:
Ah, I'll get a warning unless it's static? I'll find a place to put it, since the previous place didn't work.

Reply:
@kg So I think you will need at least two declarations.  One in src/mono/mono/metadata some place, and one in src/mono/wasm/runtime/driver.c

The way to think about the build is that the stuff in src/mono/mono doesn't know anything about src/mono/wasm.  It's  just building a  libmonosgen-2.0.a.   The src/mono/wasm/runtime code is like a completely unrelated application that happens to use libmonosgen-2.0.a. 

So a declaration in src/mono/mono/metadata/object-internals.h is so that the runtime will build without that "no previous declaration" warning/error.

The declaration in src/mono/wasm/runtime/driver.c is so the wasm code has a prototype for the function that it can use.  (the alternative is to add it to the function to the public API, or to the mono/metadata/mono-private-unstable.h API - and make sure that that header is included both in the relevant *.c files in src/mono/mono/metadata and src/mono/wasm/runtime/